pkg/parser/parser.go

```diff
@@ -27,29 +28,23 @@ func executeBootstrapTemplate(inputContract *nbcontractv1.Configuration) (string
 
 // this function will eventually take a pointer to the bootstrap contract struct.
 // it will then template out the variables into the final bootstrap trigger script.
-func Parse() {
-	inputJSON, err := json.Marshal(getBaseTemplate())
-	if err != nil {
-		log.Printf("Failed to marshal the nbcontractv1 to json: %v", err)
-	}
-
-	log.Println("Input Json: ")
-	log.Println(string(inputJSON))
-
-	// inputJson above will be provided by bootstrappers. We are using getBaseTemplate() for dev/test purpose for now.
-	// We can further move it to other file for unit tests later.
-
-	inputContract := nbcontractv1.Configuration{}
-	err = json.Unmarshal(inputJSON, &inputContract)
+func Parse(inputJSON []byte) string {
+	// Parse the JSON into a Person struct
+	var nbc nbcontractv1.Configuration
+	err := json.Unmarshal(inputJSON, &nbc)
 	if err != nil {
 		log.Printf("Failed to unmarshal the json to nbcontractv1: %v", err)
+		panic(err)
```

Comment:
Should we not panic because the program will crash immediately? I think we can return error instead so that the caller can catch this error. Thoughts?
There is a post discussing panic vs error. https://stackoverflow.com/questions/44504354/should-i-use-panic-or-return-error

Reply:
Agreed, return error instead.